processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -819,17 +678,21 @@ public Builder setContext(Map<String, Object> context)
       return this;
     }
 
+    public Builder overrideContext(Map<String, Object> contextOverride)
+    {
+      this.context = computeOverriddenContext(context, contextOverride);
+      return this;
+    }
+
     public Builder setHavingSpec(HavingSpec havingSpec)
     {
       this.havingSpec = havingSpec;
-
       return this;
     }
 
     public Builder setLimit(Integer limit)
     {
       this.limit = limit;
```

Comment:
I think this needs to clear `limitFn` and `limitSpec` in order to force them to be recomputed, otherwise code like this would ignore the `setLimit(10)` part,

```java
new GroupByQuery.Builder(query).setLimit(10).build();
```